internal/coreinternal/pdatautil/map.go

```diff
@@ -0,0 +1,90 @@
+package pdatautil
+
+import (
+	"bytes"
+	"encoding/binary"
+	"hash"
+	"sort"
+
+	"github.com/cespare/xxhash/v2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+var (
+	valEmpty        = []byte{'\xf5'}
+	valBytesPrefix  = []byte{'\xf6'}
+	valStrPrefix    = []byte{'\xf7'}
+	valBoolTrue     = []byte{'\xf8'}
+	valBoolFalse    = []byte{'\xf9'}
+	valIntPrefix    = []byte{'\xfa'}
+	valDoublePrefix = []byte{'\xfb'}
+	valMapPrefix    = []byte{'\xfc'}
+	valMapSuffix    = []byte{'\xfd'}
+	valSlicePrefix  = []byte{'\xfe'}
+	valSliceSuffix  = []byte{'\xff'}
+)
+
+// MapHash return a hash for the provided map.
+// Order of the underlying key/value pairs is ignored.
```

Comment:
if the order is ignored, does this mean the hash is not deterministic? or that we discard the order to use the internal order of elements are stored in the underlying storage.

Reply:
actually, we sort by keys line 41, so we do care about the order. Should this comment be removed?

Reply:
The order is ignored specifically to make it deterministic. I can rephrase it to make it clear

Reply:
Updated

Reply:
:+1: